plugins/reachability/src/main/java/org/qbicc/plugin/reachability/RTAInfo.java

```diff
@@ -248,6 +257,16 @@ synchronized void processReachableInstanceMethodInvoke(final MethodElement targe
                 if (!target.isPrivate()) {
                     propagateInvokabilityToOverrides(target);
                 }
+                /* Add service implementations as invokable */
+                if (isServiceSearched(definingClass)) {
+                    Set<LoadedTypeDefinition> serviceImplClasses = interfaceHierarchy.get(definingClass);
```

Comment:
I was finding that the generated service implementations did not actually contain all the necessary methods that were used in the program without some extra help. You're right though this may be looking through more classes than necessary. perhaps `searchedServiceInterfaces` should be a map to a set of found services instead to keep this list small.